front/app/containers/Admin/projects/project/ideas/AnalysisBanner.tsx

```diff
@@ -18,20 +18,35 @@ import useAddAnalysis from 'api/analyses/useAddAnalysis';
 
 const AnalysisBanner = () => {
   const { projectId } = useParams() as { projectId: string };
-  const { data: analyses } = useAnalyses({ projectId });
-  const { mutate: createAnalysis } = useAddAnalysis();
+  const { data: analyses, isLoading: isLoadingAnalyses } = useAnalyses({
+    projectId,
+  });
+  const { mutate: createAnalysis, isLoading } = useAddAnalysis();
 
   const { formatMessage } = useIntl();
 
   const analysisEnabled = useFeatureFlag({ name: 'analysis' });
 
-  useEffect(() => {
-    if (analysisEnabled && analyses && analyses.data.length === 0) {
-      createAnalysis({ projectId });
+  const handleGoToAnalysis = () => {
+    if (analyses?.data.length) {
+      clHistory.push(
+        `/admin/projects/${projectId}/analysis/${analyses?.data[0].id}`
+      );
+    } else {
+      createAnalysis(
+        { projectId },
+        {
+          onSuccess: () => {
+            clHistory.push(
+              `/admin/projects/${projectId}/analysis/${analyses?.data[0].id}`
+            );
```

Comment:
Just for my understanding: The fact that this works means that the onSuccess is only executed after all invalidations have been refetched, correct?

Reply:
You are right, we should rely on the response data here, rather than the query to have a better guarantee that the data is there. 👍